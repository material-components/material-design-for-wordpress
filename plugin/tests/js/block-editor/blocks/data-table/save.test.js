/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Save from '../../../../../assets/src/block-editor/blocks/data-table/save';

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <Save { ...props } /> );
};

const attributes = {
	hasFixedLayout: true,
	head: [
		{
			cells: [
				{
					content: 'Version',
					tag: 'th',
				},
				{
					content: 'Jazz Musician',
					tag: 'th',
				},
				{
					content: 'Release Date',
					tag: 'th',
				},
			],
		},
	],
	body: [
		{
			cells: [
				{
					content: '5.2',
					tag: 'td',
				},
				{
					content: 'Jaco Pastorius',
					tag: 'td',
				},
				{
					content: 'May 7, 2019',
					tag: 'td',
				},
			],
		},
		{
			cells: [
				{
					content: '5.1',
					tag: 'td',
				},
				{
					content: 'Betty Carter',
					tag: 'td',
				},
				{
					content: 'February 21, 2019',
					tag: 'td',
				},
			],
		},
		{
			cells: [
				{
					content: '5.0',
					tag: 'td',
				},
				{
					content: 'Bebo ValdÃ©s',
					tag: 'td',
				},
				{
					content: 'December 6, 2018',
					tag: 'td',
				},
			],
		},
	],
	foot: [
		{
			cells: [
				{
					content: 'Footer Col 1',
					tag: 'td',
				},
				{
					content: 'Footer Col 2',
					tag: 'td',
				},
				{
					content: 'Footer Col 3',
					tag: 'td',
				},
			],
		},
	],
	caption: 'Material Data Table',
	className: 'is-style-material',
};

describe( 'Data Table Save', () => {
	it( 'matches snapshot when table has data', () => {
		const wrapper = setup( { attributes } );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when table has no data', () => {
		const wrapper = setup( {
			attributes: { ...attributes, head: [], body: [], foot: [] },
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when table has rows', () => {
		const wrapper = setup( {
			attributes: { ...attributes, head: [], foot: [] },
		} );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
