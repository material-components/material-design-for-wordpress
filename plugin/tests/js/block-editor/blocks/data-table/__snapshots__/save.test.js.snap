// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Save matches snapshot when table has data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wp-block-table is-style-material"
      >
        <div
          class="mdc-data-table"
        >
          <table
            class="mdc-data-table__table has-fixed-layout"
          >
            <thead
              class=""
            >
              <tr
                class="mdc-data-table__header-row"
              >
                <th
                  class="mdc-data-table__header-cell"
                >
                  <div>
                    Version
                  </div>
                </th>
                <th
                  class="mdc-data-table__header-cell"
                >
                  <div>
                    Jazz Musician
                  </div>
                </th>
                <th
                  class="mdc-data-table__header-cell"
                >
                  <div>
                    Release Date
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="mdc-data-table__content"
            >
              <tr
                class="mdc-data-table__row"
              >
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    5.2
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    Jaco Pastorius
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    May 7, 2019
                  </div>
                </td>
              </tr>
              <tr
                class="mdc-data-table__row"
              >
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    5.1
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    Betty Carter
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    February 21, 2019
                  </div>
                </td>
              </tr>
              <tr
                class="mdc-data-table__row"
              >
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    5.0
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    Bebo Valdés
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    December 6, 2018
                  </div>
                </td>
              </tr>
            </tbody>
            <tfoot
              class=""
            >
              <tr
                class="mdc-data-table__row"
              >
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    Footer Col 1
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    Footer Col 2
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    Footer Col 3
                  </div>
                </td>
              </tr>
            </tfoot>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wp-block-table is-style-material"
    >
      <div
        class="mdc-data-table"
      >
        <table
          class="mdc-data-table__table has-fixed-layout"
        >
          <thead
            class=""
          >
            <tr
              class="mdc-data-table__header-row"
            >
              <th
                class="mdc-data-table__header-cell"
              >
                <div>
                  Version
                </div>
              </th>
              <th
                class="mdc-data-table__header-cell"
              >
                <div>
                  Jazz Musician
                </div>
              </th>
              <th
                class="mdc-data-table__header-cell"
              >
                <div>
                  Release Date
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="mdc-data-table__content"
          >
            <tr
              class="mdc-data-table__row"
            >
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  5.2
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  Jaco Pastorius
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  May 7, 2019
                </div>
              </td>
            </tr>
            <tr
              class="mdc-data-table__row"
            >
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  5.1
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  Betty Carter
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  February 21, 2019
                </div>
              </td>
            </tr>
            <tr
              class="mdc-data-table__row"
            >
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  5.0
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  Bebo Valdés
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  December 6, 2018
                </div>
              </td>
            </tr>
          </tbody>
          <tfoot
            class=""
          >
            <tr
              class="mdc-data-table__row"
            >
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  Footer Col 1
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  Footer Col 2
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  Footer Col 3
                </div>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Data Table Save matches snapshot when table has no data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Data Table Save matches snapshot when table has rows 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wp-block-table is-style-material"
      >
        <div
          class="mdc-data-table"
        >
          <table
            class="mdc-data-table__table has-fixed-layout"
          >
            <tbody
              class="mdc-data-table__content"
            >
              <tr
                class="mdc-data-table__row"
              >
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    5.2
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    Jaco Pastorius
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    May 7, 2019
                  </div>
                </td>
              </tr>
              <tr
                class="mdc-data-table__row"
              >
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    5.1
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    Betty Carter
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    February 21, 2019
                  </div>
                </td>
              </tr>
              <tr
                class="mdc-data-table__row"
              >
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    5.0
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    Bebo Valdés
                  </div>
                </td>
                <td
                  class="mdc-data-table__cell"
                >
                  <div>
                    December 6, 2018
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wp-block-table is-style-material"
    >
      <div
        class="mdc-data-table"
      >
        <table
          class="mdc-data-table__table has-fixed-layout"
        >
          <tbody
            class="mdc-data-table__content"
          >
            <tr
              class="mdc-data-table__row"
            >
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  5.2
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  Jaco Pastorius
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  May 7, 2019
                </div>
              </td>
            </tr>
            <tr
              class="mdc-data-table__row"
            >
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  5.1
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  Betty Carter
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  February 21, 2019
                </div>
              </td>
            </tr>
            <tr
              class="mdc-data-table__row"
            >
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  5.0
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  Bebo Valdés
                </div>
              </td>
              <td
                class="mdc-data-table__cell"
              >
                <div>
                  December 6, 2018
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
